### DISCUSSION TITLE
Utilizing to_spatial_smooth on a directed network

### DISCUSSION THREAD

**Discussion by jucardwell** [luukvdmeer/sfnetworks#236] [Q&A] [answered]
I am facing an issue with removing pseudo nodes from a directed road network. I have a road dataset with a significant number of pseudo nodes at areas where road characteristics change (change of speed limit, number of lanes, etc.). I have implemented the instructions in this vignette (https://cran.r-project.org/web/packages/sfnetworks/vignettes/sfn02_preprocess_clean.html) to create a directed network with two-way and one-way streets. 

However, a problem arises when I try to remove these pseudo nodes on this directed network. Since, in many situations, these pseudo nodes take place on a part of the network that has two edges (because it is a two-way street), even though this is not a true intersection, to_spatial_smooth does not remove them. 

I have provided a simplified example below. Any suggestions on how to address this? 
```
library(sfnetworks)
library(sf)
library(tidygraph)
library(tidyverse)
library(igraph)

p1 = st_point(c(7, 51))
p2 = st_point(c(7, 52))
p3 = st_point(c(8, 52))
p4 = st_point(c(7, 51.5))

l1 = st_sfc(st_linestring(c(p1, p4)))
l2 = st_sfc(st_linestring(c(p1, p3)))
l3 = st_sfc(st_linestring(c(p3, p2)))
l4 = st_sfc(st_linestring(c(p4, p2)))

edges = st_as_sf(c(l1, l2, l3, l4), crs = 4326)
edges$oneway = c(FALSE, FALSE, FALSE, FALSE)
edges

duplicates = edges[!edges$oneway, ]
reversed_duplicates = st_reverse(duplicates)

edges = rbind(edges, reversed_duplicates)
net = as_sfnetwork(edges)

edges <- st_as_sf(net, "edges")

smoothed = convert(net, to_spatial_smooth)

plot(smoothed)
```

**Comment by luukvdmeer** [comment]
Interesting, I had not thought about this. In principle, the smoothing does perform as expected (since the "non-intersection" node does have two incoming and two outgoing edges, and hence is correctly not detected as being a pseudo node). But I do see that in this case, with a network containing both oneway and two-way streets, it should in practice be treated as a pseudo node. I'll try to think of a flexible solution.

**Reply by jucardwell** [reply]
Great, thanks for your time!

**Reply by jucardwell** [reply]
Hi Luuk! Still working on this issue, and have been unsuccessful so far. Have you had any ideas?

**Comment by luukvdmeer** [comment, marked as answer]
From sfnetworks v1.0, you can work around this by first creating the network, cleaning it, and only then creating the "mixed network" (i.e. the network with both directed and undirected edges through the duplicate-and-reverse workflow) using the new morpher `to_spatial_mixed()`:

```r
as_sfnetwork(roxel) |>
  convert(to_spatial_smooth) |>
  convert(to_spatial_mixed, directed = oneway)
```
